apps/shared/avifutil.h

```diff
@@ -10,6 +10,59 @@
 extern "C" {
 #endif
 
+// NEXTARG(), MAIN(), INIT_ARGV(), FREE_ARGV() for UTF8-aware command line parsing.
+#if defined(_WIN32)
+#define NEXTARG()                                                      \
+    if (((argIndex + 1) == argc) || (argv[argIndex + 1][0] == L'-')) { \
+        fprintf(stderr, "%s requires an argument.", arg);              \
+        goto cleanup;                                                  \
+    }                                                                  \
+    arg = argv[++argIndex]
+#else
+#define NEXTARG()                                                     \
+    if (((argIndex + 1) == argc) || (argv[argIndex + 1][0] == '-')) { \
+        fprintf(stderr, "%s requires an argument.", arg);             \
+        goto cleanup;                                                 \
+    }                                                                 \
+    arg = argv[++argIndex]
+#endif
+
+#if defined(_WIN32)
+#define MAIN() int wmain(int argc, wchar_t * _argv[])
+#else
+#define MAIN() int main(int argc, char * argv[])
+#endif
+
+#if defined(_WIN32)
+#define INIT_ARGV()                                                                                           \
+    char * argvAll = NULL;                                                                                    \
+    char ** argv = NULL;                                                                                      \
+    if (setlocale(LC_ALL, ".UTF8") == NULL) {                                                                 \
+        fprintf(stderr, "setlocale failed\n");                                                                \
+        goto cleanup;                                                                                         \
+    }                                                                                                         \
+    argvAll = (char *)malloc(1024 * argc * sizeof(*argvAll));                                                 \
+    argv = (char **)malloc(argc * sizeof(*argv));                                                             \
+    for (int i = 0; i < argc; ++i) {                                                                          \
+        argv[i] = argvAll + 1024 * i;                                                                         \
+        int rc = WideCharToMultiByte(CP_UTF8, WC_ERR_INVALID_CHARS, _argv[i], -1, argv[i], 1024, NULL, NULL); \
+        if (rc == 0) {                                                                                        \
+            fprintf(stderr, "WideCharToMultiByte() failed\n");                                                \
+            goto cleanup;                                                                                     \
```

Comment:
Optional: Since we should ideally call `INIT_ARGV()` first thing in the main() function, we can probably just free `argv` and `argvAll` and return 1, instead of goto cleanup.

This will allow us to avoid moving the `decoded` array before `INIT_ARGV()` in tests/gtest/are_images_equal.cc.